src/mmw/js/src/core/views.js

```diff
@@ -79,11 +80,25 @@ var HeaderView = Marionette.ItemView.extend({
     },
 
     templateHelpers: function() {
-        var self = this;
+        var self = this,
+            currentActive = self.appState.get('active_page'),
+            isSelectArea = currentActive === coreUtils.selectAreaPageTitle,
+            isProjects = currentActive === coreUtils.projectsPageTitle,
+            isAnalyze = currentActive === coreUtils.analyzePageTitle,
+            modelPackageNames = _.pluck(settings.get('model_packages'), 'display_name'),
+            isModel = _.contains(modelPackageNames, currentActive),
+            isCompare = currentActive === coreUtils.comparePageTitle,
+            wasAnalyzeVisible = self.appState.get('was_analyze_visible'),
+            wasModelVisible = self.appState.get('was_model_visible'),
+            wasCompareVisible = self.appState.get('was_compare_visible');
 
         return {
             'itsi_embed': settings.get('itsi_embed'),
-            'current_page_title': self.appState.get('current_page_title')
+            'selectAreaNavClasses': (isSelectArea ? 'active' : '') + (isProjects ? '' : ' visible'),
+            'analyzeNavClasses': (isAnalyze ? 'active' : '') + (wasAnalyzeVisible || wasModelVisible ? ' visible' : ''),
+            'modelNavClasses': (isModel ? 'active' : '') +  (isModel || isCompare ? ' visible' : ''),
+            'compareNavClasses': (isCompare ? 'active' : '') +
+                ((isModel && wasCompareVisible) || isCompare ? ' visible' : ''),
```

Comment:
Case statements could just return different combinations of these options? One one hand it'd probably lead to some code repetition; on the other I think it would make it more apparent which switches do what in this method.